doc/examples/reconst_dki.py

```diff
@@ -48,99 +53,99 @@
+taken into account by the diffusion kurtosis tensor), we only select the
+b-values up to 2000 $s.mm^{-2}$:
 """
 
-from dipy.data import read_sherbrooke_3shell
+bvals = [200, 400, 1000, 2000]
 
-img, gtab = read_sherbrooke_3shell()
+img, gtab = read_cenir_multib(bvals)
 
 data = img.get_data()
 
+affine = img.get_affine()
+
 """
-img contains a nibabel Nifti1Image object (with the data) and gtab contains a
-GradientTable object with information about the b-values and b-vectors. The
-b-values used on the loaded dataset are visualized below
+Function ``read_cenir_multib`` return img and gtab which contains respectively
+a nibabel Nifti1Image object (where the data can be extracted) and a
+GradientTable object with information about the b-values and b-vectors.
+
+Before fitting the data, we preform some data pre-processing. We first create a
+preliminary mask to avoid preprocessing unnecessary voxels:
 """
 
-import matplotlib.pyplot as plt
-
-plt.plot(gtab.bvals, label='b-values')
-fig1 = plt.gcf()
-plt.legend()
-plt.show()
-fig1.savefig('HARDI193_bvalues.png')
+maskdata, mask = median_otsu(data, 4, 2, False, vol_idx=[0, 1], dilate=1)
 
 """
-.. figure:: HARDI193_bvalues.png
-   :align: center
-   **b-values of the loaded dataset**.
-
-From the figure above, we can check that the loaded dataset contains three
-non-zero b-values as required for DKI. However the highest b-value of 3500
-$s.mm^{-2}$ is higher than normally used on DKI. Since DKI neglects diffusion
-signal components higher than the 4th order KT, a upper bound of b-value < 3000
-$s.mm^{-2}$ is normally implied to decrease the influence of higher order terms
-[Jensen2010]_. Following this, we discard the b-value shell of 3500 $s.mm^{-2}$
-before DKI fitting.
+Since the diffusion kurtosis models involves the estimation of a large number
+of parameters [TaxCMW2015]_ and since the non-Gaussian components of the
+diffusion signal are more sensitive to artefacts [NetoHe2012]_, a fundamental
+data pre-processing step for diffusion kurtosis fitting is to denoise our data.
+For this, we use Dipy's non-local mean filter (see
+:ref:`example-denoise-nlmeans`). Note that, since the HCP-like data has a large
+number of diffusion-weigthed volume, this procedure can take a couple of hours
+to compute.
 """
 
-select_ind = gtab.bvals < 3000
-selected_bvals = gtab.bvals[select_ind]
-selected_bvecs = gtab.bvecs[select_ind, :]
-selected_data = data[:, :, :, select_ind]
+sigma = estimate_sigma(data, N=4)
+den = nlmeans(data, sigma=sigma, mask=mask)
 
 """
-The selected b-values and gradient directions are then converted to Dipy's
-GradientTable format.
+To avoid future reprocessing of the non-local mean field, below we save the
+denoised version of the HCP-like data.
 """
 
-from dipy.core.gradients import gradient_table
-
-gtab = gradient_table(selected_bvals, selected_bvecs)
+nib.save(nib.Nifti1Image(den, affine), 'denoised_cenir_multib.nii.gz')
```

Comment:
You can use a mask to define a roi, it should only process those voxels and
give the same thing as the full one.
On Sep 9, 2015 6:06 AM, "Rafael Neto Henriques" <notifications@github.com>
wrote:

> In doc/examples/reconst_dki.py
> <https://github.com/nipy/dipy/pull/677#discussion_r39026002>:
>
> >  """
> >
> > -from dipy.core.gradients import gradient_table
> > -
> > -gtab = gradient_table(selected_bvals, selected_bvecs)
> > +nib.save(nib.Nifti1Image(den, affine), 'denoised_cenir_multib.nii.gz')
>
> Hi @samuelstjean <https://github.com/samuelstjean>! I didn't work on a
> crop of the data because this will affect the performance of the nlmeans,
> right? Regarding to @ChantalTax <https://github.com/ChantalTax>
> suggestion of saving this dataset online, see the comments done below
> (particularly the comment done at September 6).
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/nipy/dipy/pull/677/files#r39026002>.
>